packages/mobile/src/exchange/saga.ts

```diff
@@ -300,15 +320,15 @@ export function* exchangeGoldAndStableTokens(action: ExchangeTokensAction) {
 
     // Generate and send a transaction to approve payment to the exchange.
     let approveTx
-    if (makerToken === Currency.Celo) {
+    if (!isStableToGold) {
       approveTx = goldTokenContract.approve(
         exchangeContract.address,
         convertedMakerAmount.toString()
       )
     } else {
       approveTx = stableTokenContract.approve(
         exchangeContract.address,
-        convertedMakerAmount.toString()
+        convertedTakerAmount.multipliedBy(10 ** 20).toString()
```

Comment:
Why are you multiplying by this number here?

Reply:
ops! it was just a test, I'm going to clean it